arch/arm/core/aarch32/cpu_idle.S

```diff
@@ -73,6 +73,12 @@ SECTION_FUNC(TEXT, arch_cpu_idle)
 #error Unknown ARM architecture
 #endif /* CONFIG_ARMV6_M_ARMV8_M_BASELINE */
 
+	/* Synchronization barriers before calling WFI
+	* DSB is recommented by architecture definitions
+	* ISB flushes the instruction pipeline after setting PRIMASK/BASEPRI
+	*/
+	dsb
+	isb
```

Comment:
Not sure if `ISB` is necessary/required here.

Reply:
Note that in the context of #23511, `CPSID` is guaranteed to be self synchronising and using `ISB` is not required in this particular instance.

Reply:
agreed about CPSID, but not when setting BASEPRI - this needs ISB